File: src/main/java/com/scottscmo/Application.java
package com.scottscmo;

import com.formdev.flatlaf.FlatLightLaf;
import com.scottscmo.ui.panels.BibleInfoPanel;
import com.scottscmo.ui.panels.GSlidesPanel;
import com.scottscmo.ui.panels.PPTXGeneratorsPanel;
import com.scottscmo.ui.panels.SettingsPanel;
import com.scottscmo.ui.panels.SongFormatterPanel;
import net.miginfocom.swing.MigLayout;

import javax.swing.JFrame;
import javax.swing.JTabbedPane;
import javax.swing.SwingUtilities;
import java.awt.LayoutManager;

public class Application extends JFrame {

    private static Application _app;
    public static Application get() {
        return _app;
    }

    public Application() {
        setTitle("Worship Service Tool");
        setDefaultCloseOperation(EXIT_ON_CLOSE);
        setResizable(false);

        LayoutManager layout = new MigLayout("ins 0, wrap");
        getContentPane().setLayout(layout);

        JTabbedPane tabs = new JTabbedPane();
        tabs.addTab("Bible", new BibleInfoPanel());
        tabs.addTab("Song Formatter", new SongFormatterPanel());
        tabs.addTab("PPTX Generators", new PPTXGeneratorsPanel());
        tabs.addTab("Google Slides", new GSlidesPanel());
        tabs.addTab("Settings", new SettingsPanel());
        getContentPane().add(tabs);

        pack(); // auto-resize to component, use setSize if need fixed size
    }

    public static void main(String[] args) {
        Thread.setDefaultUncaughtExceptionHandler((Thread t, Throwable e) -> {
            AppLogger.error("Uncaught exception in thread: " + t.getName(), e);
        });

        Config.reload();
        FlatLightLaf.setup();
        SwingUtilities.invokeLater(() -> {
            _app = new Application();
            _app.setVisible(true);
        });
    }
}

File: src/main/kotlin/com/scottscmo/Application.kt
package com.scottscmo

import com.formdev.flatlaf.FlatLightLaf
import com.scottscmo.ui.OutputDisplay
import com.scottscmo.ui.components.DataPathPicker
import com.scottscmo.ui.panels.*
import net.miginfocom.swing.MigLayout
import javax.swing.JFrame
import javax.swing.JTabbedPane
import javax.swing.SwingUtilities

class Application() : JFrame() {

    init {
        title = "Worship Service Tool"
        defaultCloseOperation = EXIT_ON_CLOSE
        isResizable = false

        contentPane.apply {
            layout = MigLayout("ins 0, wrap")
            add(DataPathPicker().ui)
            add(JTabbedPane().apply {
                addTab("Bible", BibleInfoPanel())
                addTab("Song Formatter", SongFormatterPanel())
                addTab("PPTX Generators", PPTXGeneratorsPanel())
                addTab("Google Slides", GSlidesPanel())
                addTab("Settings", SettingsPanel())
            })
        }

        pack() // auto-resize to component, use setSize if need fixed size
    }

    companion object {
        private lateinit var app: Application
        fun get(): Application {
            return app
        }
        @JvmStatic
        fun main(args: Array<String>) {
            Config.load()
            FlatLightLaf.setup()
            SwingUtilities.invokeLater {
                val defaultHandler = Thread.getDefaultUncaughtExceptionHandler()
                Thread.setDefaultUncaughtExceptionHandler { thread, throwable ->
                    defaultHandler?.uncaughtException(thread, throwable)
                    // show uncaught error to user
                    OutputDisplay.error(throwable.message + "\n"
                        + throwable.stackTrace.copyOfRange(0, 10).joinToString("\n"))
                }

                app = Application().apply { isVisible = true }
            }
        }
    }
}

